lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -317,9 +317,12 @@ sub prepare_complex_query_search_args ($self, $args) {
             push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
         }
 
-        for my $key (qw(build distri version flavor arch test machine)) {
+        for my $key (qw(distri version flavor arch test machine)) {
             push(@conds, {"me." . uc($key) => $args->{$key}}) if $args->{$key};
         }
+        if (defined $args->{build}) {
+            push(@conds, {"me.BUILD" => {-in => $args->{build}}});
```

Comment:
Ok, so apparently `$args->{build}` is an array ref here. I hope that's always¹ the case and that the values contained by that array ref are sufficiently validated.

¹ This function might be called in multiple places.

Reply:
ARRAYREF or Scalar .. it comes from `_compose_job_overview_search_args` if isn't defined in query or from query ..

Reply:
But we call `complex_query` (which calls this function) also in other places, e.g. `OpenQA::WebAPI::Controller::API::V1::Job::list`. Have you checked the other occurrences as well?

Reply:
hmm, you're right

Reply:
@Martchus probadly overkill but in this case better safe than sorry 🗡️